common/lib/xmodule/xmodule/js/src/video/01_initialize.js

```diff
@@ -286,7 +286,7 @@ function(VideoPlayer, i18n, moment, _) {
      */
     function extractHLSVideoSources(state) {
         return _.filter(state.config.sources, function(source) {
-            return /\.m3u8$/.test(source);
+            return /\.m3u8(?.*)?$/.test(source);
```

Comment:
You need to escape the first question mark.  As written now, the regex is invalid and results in a syntax error exception.